_drafts/release-workflows.md

```diff
@@ -0,0 +1,388 @@
+| **push** to `5.0.0-beta`                                       | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+| **merge** any branch to `5.0.0-beta`                           | Release a prerelease version (`5.0.0-beta`, then `5.0.0-beta.1`, then `5.0.0-beta.2`) on default channel (if supported, otherwise no channel) |
+
+##### Pre-releases configuration
+
+```json
+"release": {
+  "branches": ["master", "4.0.0-beta", "5.0.0-alpha"]
+}
+```
+
+With specific branch names:
+
+```json
+"release": {
+  "branches": ["master", {
+    "branch": "dev",
+    "prerelease": "4.0.0-alpha"
+  }, {
+    "branch": "experimental",
+    "prerelease": "5.0.0-beta"
+  }]
+}
+```
+
+#### Configuration
+
+##### Format
+
+Each branch is either a `String` or an `Object` with the following properties:
+
+| Option       | Description                                                                                                                                                             |
+|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `branch`     | **Require** Git branch name                                                                                                                                             |
+| `range`      | Accepted version range for to release from this branch. If defined the branch will be considered a LTS branch.                                                          |
+| `channel`    | Channel on which to release. Ignored for release targets that don't support channels.                                                                                 |
+| `prerelease` | Version of the pre-release done from this branch. Must be formatted `<version>-<tag>`. The version is required, the `tag` is optional. Examples: `4.0.0-beta`, `4.0.0`. |
+
+**The order in which the branches are defines is meaningful, and it is used to determine the branch type and which release can be done from each branch.**
+
+##### Branches naming convention
+
+In a branch is defined as a `String` `semantic-release` will automatically determine the type of branch as follow:
+- If the branch name is a valid semver range (`1.x.x`, `1.0.x`, `^1.0.0`) the branch will be considered a LTS branch, with its range based on the branch name, and release on the channel with the same name formatted `1.x.x` (`^1.0.0` => `1.x.x`)
+- The first branch in the `branches` `Array` after the LTS branches is considered the default branch (usually `master`).
+- Each branches in the `branches` `Array` after the default branch are considered future branches.
+- If a future branch is named `minor`, `major`, `breaking` or `feature` it's type will be determined accordingly
```

Comment:
Can you further explain what this means? I can’t think of a good reason where I would call a branch "breaking"

Reply:
It's because `breaking` is synonym of `major`. That would create a branch that is meant to work on the next breaking (or major) releasing.

This is just becasue we want to push the idea that:
- `fix === patch`
- `feature === minor`
- `major === breaking`

Reply:
I understand that `breaking` is an alias to `major`, i just don’t see a use case where I would create a branch with such names, it’s just not something I have seen and I wouldn’t need if we support `next` as a special branch name

Reply:
Ok. So that means to stay consistent we would also not auto-determine anything for branches called `feature`

Reply:
yes exactly, I would remove any support for branches called `fix`/`patch`, `feature`/`minor` or `breaking`/`major`

Reply:
Why would we do that?
That allow user to avoid configuration by detecting the type of branch automatically, the same way automatically determine `1.x` is a LTS branch to maintain.
Everywhere in this specs we try to allow users to express their intent with branch naming. I don't understand why we would make an exception here.

Reply:
I don’t think we need the distinction between the type of change between the main branch and the future branch. And I want to embrace existing conventions just like we did with the angular commit conventions back in the days. I’ve never seen branches called `minor` or `major`, it’s a new concept we would need to explain and users of semantic-release would need to explain to their users and collaborators